rewatch/src/mcp.rs

```diff
@@ -0,0 +1,205 @@
+use anyhow::Result;
+use mcp_server::router::{CapabilitiesBuilder, Router, RouterService};
+use mcp_server::{ByteTransport, Server};
+use mcp_spec::{
+    content::Content, prompt::Prompt, protocol::ServerCapabilities, resource::Resource, tool::Tool,
+};
+use serde_json::json;
+use std::path::Path;
+use std::pin::Pin;
+use std::process::Command;
+
+use crate::{build, helpers, project_context::ProjectContext};
+
+#[derive(Clone)]
+pub struct RewatchMcp;
+
+impl RewatchMcp {}
+
+impl Router for RewatchMcp {
+    fn name(&self) -> String {
+        "rescript-mcp".to_string()
+    }
+
+    fn instructions(&self) -> String {
+        "ReScript MCP server.\n\n\
+        Tools:\n\
+        - diagnose(path): Quick per-file diagnostics after edits. Prefer this before a full build.\n\
```

Comment:
This tool call does feel a bit like duplication of one of the LSP endpoints.
It makes me wonder if we should explore having an LSP client inside the MCP tool that can ask this information from the running LSP Server.